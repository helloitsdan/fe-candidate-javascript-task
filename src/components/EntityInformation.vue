<template>
  <div>
    <h4>
      <strong>{{ name }}</strong
      >, a {{ breed.toLowerCase() }} {{ species.toLowerCase() }}
    </h4>
    <p class="is-marginless">
      Born on {{ date_of_birth | moment("dddd, MMMM Do YYYY") }}
    </p>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "EntityInformation",
  props: {
    name: String,
    species: String,
    breed: String,
    date_of_birth: String
  }
});
</script>
