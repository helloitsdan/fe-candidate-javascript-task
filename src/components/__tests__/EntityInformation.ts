import { shallowMount } from "@vue/test-utils";

import EntityInformation from "../EntityInformation.vue";

describe("EntityInformation", () => {
  const fakeEntity = {
    name: "Noriko",
    species: "CAT",
    breed: "MIXED",
    date_of_birth: "2019-07-13T02:57:55.235Z",
  };

  it("renders", () => {
    const wrapper = shallowMount(EntityInformation, {
      propsData: fakeEntity
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
