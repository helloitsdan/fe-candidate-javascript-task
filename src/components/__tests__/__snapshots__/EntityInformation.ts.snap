// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityInformation renders 1`] = `
<div>
  <h4>
    <strong>
      Noriko
    </strong>
    , a mixed cat
  
  </h4>
   
  <p
    class="is-marginless"
  >
    
    Born on Saturday, July 13th 2019
  
  </p>
</div>
`;
